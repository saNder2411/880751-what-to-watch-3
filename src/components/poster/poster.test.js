import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {BrowserRouter, Switch, Route} from 'react-router-dom';
import Poster from './poster.jsx';

const mockStore = configureStore();
const store = mockStore({
  genre: `All Genre`,
  cardsData: [],
  reviews: [],
  newReviews: [],
  promoCardData: {},
});

const poster = `the-grand-budapest-hotel-poster`;


describe(`Render Poster`, () => {
  it(`Should Poster render correctly in CardScreen`, () => {
    const markup = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter>
              <Switch>
                <Route
                  path='/cards:id'
                  render={() => {
                    return <Poster poster={poster} isCardScreen/>;
                  }}
                />
              </Switch>
            </BrowserRouter>
          </Provider>
      )
      .toJSON();

    expect(markup).toMatchSnapshot();
  });

  it(`Should Poster render correctly in Main`, () => {
    const markup = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter>
              <Switch>
                <Route
                  path='/'
                  render={() => {
                    return <Poster poster={poster} />;
                  }}
                />
              </Switch>
            </BrowserRouter>
          </Provider>
      )
      .toJSON();

    expect(markup).toMatchSnapshot();
  });
});

