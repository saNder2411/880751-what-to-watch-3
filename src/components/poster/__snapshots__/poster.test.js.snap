// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Poster Should Poster render correctly in CardScreen 1`] = `
<div
  className="movie-card__poster movie-card__poster--big"
>
  <img
    alt="Terminator"
    height="327"
    src="the-grand-budapest-hotel-poster"
    width="218"
  />
</div>
`;

exports[`Render Poster Should Poster render correctly in Main 1`] = `
<div
  className="movie-card__poster "
>
  <img
    alt="Terminator"
    height="327"
    src="the-grand-budapest-hotel-poster"
    width="218"
  />
</div>
`;
